pyccel/codegen/printing/pycode.py

```diff
@@ -151,8 +152,73 @@ def _find_functional_expr_and_iterables(self, expr):
                 raise NotImplementedError("Type {} not handled in a FunctionalFor".format(type(body)))
         return body, iterables
 
+    def _get_numpy_name(self, expr):
+        """
+        Get the name of a NumPy function and ensure it is imported.
+
+        Get the name of a NumPy function from an instance of the class. The
+        name is saved in the class by default, however _aliases are checked
+        in case the function was imported explicitly with a different name
+        (e.g. `from numpy import int32 as i32`). If the name is not found in
+        aliases then it is added to the objects imported from NumPy.
+
+        Parameters
+        ----------
+        expr : PyccelInternalFunction
+            A pyccel node describing a NumPy function.
+
+        Returns
+        -------
+        str
+            The name that should be used in the code.
+        """
+        if isinstance(expr, type):
+            cls = expr
+        else:
+            cls = type(expr)
+        type_name = expr.name
+        name = self._aliases.get(cls, type_name)
+        if name == type_name:
+            self.insert_new_import(
+                    source = 'numpy',
+                    target = AsName(cls, name))
+        return name
+
     #----------------------------------------------------------------------
 
+    def _print_dtype_argument(self, expr, init_dtype):
+        """
+        Print a dtype argument.
+
+        Print the argument `dtype=X` from the dtype initially provided.
+
+        Parameters
+        ----------
+        expr : PyccelAstNode
+            The expression whose datatype is being determined.
+
+        init_dtype : PythonType, PyccelFunctionDef, LiteralString, str
+            The actual dtype passed to the NumPy function.
+
+        Returns
+        -------
+        str
+            The code for the dtype argument.
+        """
+        if init_dtype is None:
+            return ''
+        elif isinstance(init_dtype, (PythonType, NumpyResultType)):
+            dtype = self._print(init_dtype)
+            return "dtype = " + dtype
+        elif isinstance(init_dtype, PyccelFunctionDef):
+            dtype = self._get_numpy_name(init_dtype.cls_name)
+            return f"dtype={dtype}"
+        else:
+            dtype = self._print(expr.dtype)
+            if expr.precision != -1:
+                dtype = self._get_numpy_name(DtypePrecisionToCastFunction[datatype(dtype).name][expr.precision])
+            return f"dtype={dtype}"
```

Comment:
I think that each of the three returned strings should read
```python
return f"dtype = {dtype}"
```
We could reduce code duplication as follows:
```suggestion

        if isinstance(init_dtype, (PythonType, NumpyResultType)):
            dtype = self._print(init_dtype)
        elif isinstance(init_dtype, PyccelFunctionDef):
            dtype = self._get_numpy_name(init_dtype.cls_name)
        else:
            dtype = self._print(expr.dtype)
            if expr.precision != -1:
                dtype = self._get_numpy_name(DtypePrecisionToCastFunction[datatype(dtype).name][expr.precision])
        return f"dtype = {dtype}"
```

Reply:
Fixed [7eca256](https://github.com/pyccel/pyccel/pull/1260/commits/7eca2561d3958f04f652c1061b62c56ea9c92ef9)